blocks/header/header.js

```diff
@@ -125,41 +223,232 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
+        navSection.parentElement.setAttribute('aria-expanded', expanded ? 'false' : 'true');
+      } else {
+        // For Mobile need to set the right subsection to active so it display
+        document.getElementById(`sub-menu-${event.target.id}`).classList.toggle('sub-menu-section-active');
+
+        // Turn on the Back to Main menu button
+        const back = document.getElementById('back-button');
+        back.classList.add('nav-back-button-active');
+        // This seems like a total hack but is the only way I can get this animation to work
+        // By triggering it after a brief delay to allow the display:none on the menu to change
+        // eslint-disable-next-line no-promise-executor-return
+        await new Promise((r) => setTimeout(r, animationDelay));
+        back.classList.add('nav-back-button-animation-start');
+
+        // Transition all of the other nav elements off to the left
+        document.querySelector('header.header-wrapper').classList.add('sub-menu-selected');
+
+        // Render a subsection text to let the user know what menu item they are currently on
+        document.getElementById(`${event.target.id}-label`).classList.add('sub-menu-label-active');
+      }
+    });
+  });
+}
+
+/**
+ * Build the hamburger icon element which will trigger the menu display in mobile mode
+ * @returns {HTMLDivElement} Div containing the close button with event listeners
+ * @param {Element} nav The container element
+ * @param {Element} navSections The nav sections within the container element
+ */
+function buildHamburger(nav, navSections) {
+  const hamburger = document.createElement('div');
+  hamburger.classList.add('nav-hamburger');
+  hamburger.innerHTML = `<button type="button" aria-controls="nav" aria-label="Open navigation">
+        <span class="nav-hamburger-icon"></span>
+      </button>`;
+
+  hamburger.addEventListener('click', () => toggleMenu(nav, navSections));
+  hamburger.addEventListener('click', () => toggleNavBackground(nav.attributes.getNamedItem('aria-expanded').value === 'false'));
+  return hamburger;
+}
+
+/* ------------------------------ Main function invoked at load ------------------------------- */
+
+/**
+ * Build the hamburger icon element which will trigger the menu display in mobile mode
+ * @returns {HTMLDivElement} Div containing the return to main menu button with event listeners
+ */function buildBackButton() {
+  // Back button for mobile
+  const backButton = createDiv('back-button', ['nav-back-button']);
+  backButton.innerHTML = `<button type="button" aria-controls="nav" aria-label="back to site navigation menu">
+        Main Menu
+      </button>`;
+  backButton.addEventListener('click', () => toggleSubMenu());
+  return backButton;
+}
+
 /**
  * decorates the header, mainly the nav
  * @param {Element} block The header block element
  */
-// TODO: Split this function up breaking out to additional functions for ease of reading
 export default async function decorate(block) {
```

Comment:
Basically the decorate, decorateSection and decorateSubSection methods are all in my opinion a little overly complex and hard to read.  Probably worth a bit of refactoring to simplify and make things a bit more reusable.  Didn't want to keep whacking away on this PR for much longer though, so submitting it in the state it's in for now.